docs/RFCs/personal-data-syncing.md

```diff
@@ -0,0 +1,167 @@
+## Summary
+
+This RFC proposes usage of IPFS-Cluster in conjunction with two separate components on a BOX to sync files and folders across several different BOXes.
+
+## Use Case
+
+A person owns several BOXes and they add a document to one of them.
+
+The document immediately gets transferred over to the other BOX so that they can make use of it (ie/ perform CRUD operations on it) from either BOX.
+
+The 'documents' being synced (for now) are:
+
+    * plaintext (eg/ UTF-8 encoded) files
+
+    * photos
+
+    * video
+
+
+## Out of Scope
+
+* high-frequency updates to documents which might be encountered with multi-tenancy or recording streaming video
+
+* data persisted by orbit db or any other dbs
+
+## Pre-conditions
+[pre-conditions]: #pre-conditions
+
+* the owner has already authenticated themselves with each BOX
+
+* the fula-api is persisting data to a disk accessible by the correspondant and newsroom processes
+
+## Design Considerations
+
+### IPFS-Cluster
+
+A properly configured ipfs-cluster provides most of the properties required to implement this use case including.
+
+  * a private network of trusted peers
+
+  * conflict resolution
+
+  * notifying peers of changes
+
+  * efficient transmission of data
+
+  * providing peer pinning status
+
+### Conflict Resolution
+
+IPFS Cluster provides [two choices](https://cluster.ipfs.io/documentation/guides/consensus/) for conflict resolution.
+
+CRDT will be used because:
+
+  * it utilizes 'GossipSub' which will probably be useful for future use cases related to replication.
+
+  * it does not require > 50% of peers to be online in order to operate
+
+  * 'follower' peers might be used in the future to provide 'read-only' access
+
+  * 'trusted peers' might be used in the future to provide access control
+
+  * peers will probably come and go frequently in the future when data is synced with external devices (ie/ owner's computer or phone)
+
+  * since we have full control over the document store, we can ensure the DAG is shallow (based on limitations of number of files in a directory)
+
+  * less stable / tested conflict resolution can be tolerated for this use case
+
+## Design
+### BOX System Architecture
+
+The following diagram illustrates how  various components on the BOX will work together to implement data syncing.
+
+There are two new components introduced:
+
+  * correspondant
+
+  * newsroom
+
+![Box System Architecture](/diagrams/BOX-data-sync-architecture.png)
+
+### System Properties
+
+  * peers should not enter a recursive loop with each other by notifying authors of updates that originated from them
+
+  * each program should be reentrant in the sense that they can be aborted at any point during execution then restarted
+
+     * the shared state will be updated to reflect the latest changes to the file system
+
+
+### Correspondant
+
+The role of the correspondant is to ensure IPFS-Cluster is pinning the latest files created and updated on the author's file system.
```

Comment:
### For upload Scenario:
The @functionland/fula API Get a file and return a CID to user.
And in current implementation of [@functionland/box](https://github.com/functionland/fula/blob/main/apps/box/src/file/index.ts) we use js-ipfs to store stream and return the CID to the user. 
So can you describe the change needed to add to  @functionland/box to work.